hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java

```diff
@@ -469,4 +456,13 @@ public AtomicLong getAttemptCounter() {
   public int getThreadNo() {
     return threadNo;
   }
+
+  protected OzoneClient createOzoneClient(String omServiceID,
+      OzoneConfiguration conf) throws Exception {
+    if (omServiceID != null) {
+      return OzoneClientFactory.getRpcClient(omServiceID, conf);
+    } else {
+      return OzoneClientFactory.getRpcClient(conf);
```

Comment:
I think here `getClient` should be used to validate presence of `--om-service-id` param.

```suggestion
      return OzoneClientFactory.getClient(conf);
```

There are two methods in `OzoneClientFactory` that take `Configuration` and return `OzoneClient`:

 * `getClient(Configuration)` throws exception if configuration has OM service ID, but
 * `getRpcClient(Configuration)` does not

I'm getting timeout on OM HA cluster when `--om-service-id` is omitted.

```
$ ozone freon ockg -n 1 -t 1
...
[main] INFO ipc.Client: Retrying connect to server: 0.0.0.0/0.0.0.0:9862. Already tried 0 time(s); retry policy is RetryUpToMaximumCountWithFixedSleep(maxRetries=10, sleepTime=1000 MILLISECONDS)
...
[main] ERROR ha.OMFailoverProxyProvider: Failed to connect to OMs: [nodeId=null,nodeAddress=0.0.0.0:9862]. Attempted 15 failovers.
```

After making the suggested change locally, I get the error message as expected:

```
$ ozone freon ockg -n 1 -t 1
...
Following ServiceID's [id1] are defined in the configuration. Use the method getClient which takes serviceID and configuration as param
```

Reply:
Removed getClient, and used getRpcClient, to avoid these kind of mistakes instead of duplicating the same code in multiple functions.